src/main.cpp

```diff
@@ -40,22 +41,25 @@ int main(int argc, char **argv) {
   ros::Rate update_rate(update_rate_hz);
   ros::Time ts = ros::Time::now();
   while (!shouldShutdown) {
-    forqueSensorHW.update();
+    errorShutdown = !forqueSensorHW.update();
     ros::Duration d = ros::Time::now() - ts;
     ts = ros::Time::now();
     cm.update(ts, d);
     update_rate.sleep();
+    shouldShutdown = errorShutdown;
```

Comment:
What happens if `shouldShutdown` was set to true by the sigint handler and `errorShutdown` was false? Should this be something like `shouldShutdown |= errorShutdown`?